legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java

```diff
@@ -169,84 +198,123 @@ protected void serializeOutput(MutableMap<FileGenerationSpecification, List<Gene
         getLog().info(String.format("Done serializing %,d file generations' output to %s (%.9fs)", generationGenerationOutputMap.size(), this.outputDirectory, (System.nanoTime() - serializeStart) / 1_000_000_000.0));
     }
 
-    private void filterGenerationSpecsByIncludes(Map<String, GenerationSpecification> generationSpecsByPath) throws MojoExecutionException
+
+    protected void serializeArtifacts(MutableMap<ArtifactGenerationExtension, List<ArtifactGenerationResult>> results, Set<String> fileOutputPaths, PureModel pureModel) throws IOException, MojoExecutionException
+    {
+        long serializeStart = System.nanoTime();
+        getLog().info("Start serializing artifact extension generations");
+        Path outputDirPath = this.outputDirectory.toPath();
+        Pattern pkgSepPattern = Pattern.compile("::", Pattern.LITERAL);
+        String replacement = Matcher.quoteReplacement(outputDirPath.getFileSystem().getSeparator());
+        results.forEachKeyValue((extension, extensionResults)  ->
+        {
+            for (ArtifactGenerationResult result: extensionResults)
+            {
+                org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.PackageableElement generator = result.getElement();
+                String elementFolder = org.finos.legend.pure.m3.navigation.PackageableElement.PackageableElement.getUserPathForPackageableElement(generator, FILE_PATH_SEPARATOR);
+                List<GenerationOutput> generatorOutputs = result.getResults();
+                String rootExtensionFolder = extension.getKey();
+                for (GenerationOutput output: generatorOutputs)
+                {
+                    try
+                    {
+                        String fileName = elementFolder + FILE_PATH_SEPARATOR + rootExtensionFolder + FILE_PATH_SEPARATOR + output.getFileName();
+                        if (!fileOutputPaths.add(fileName))
+                        {
+                            throw new MojoExecutionException("Duplicate file path found when serializing artifact generation extension  '" + extension.getClass() + "' output: '" + fileName + "'");
+                        }
+                        String resolver = pkgSepPattern.matcher(fileName).replaceAll(replacement);
+                        Path entityFilePath = outputDirPath.resolve(resolver);
+                        Files.createDirectories(entityFilePath.getParent());
+                        try (OutputStream stream = Files.newOutputStream(entityFilePath))
+                        {
+                            stream.write(output.extractFileContent().getBytes(StandardCharsets.UTF_8));
+                        }
+                    } catch (Exception exception)
+                    {
+                        throw new RuntimeException(exception);
```

Comment:
Do not just blindly wrap exceptions in RuntimeException.

Reply:
as discussing using a for each instead of forEachKeyValue